src/controllers/userController.js

```diff
@@ -85,14 +76,56 @@ const login = async (req, res) => {
 
   const data = {
     user: {
+      id: foundUser.id,
       email: foundUser.email,
       token: authHelper.generateToken({ id: foundUser.id }),
-      username: foundUser.username,
       bio: foundUser.bio
     }
   };
   return successResponse(res, 200, data);
 };
 
+/**
+ * @description updates the profile of a user
+ * @param {object} request express request object
+ * @param {object} response express response object
+ * @param {object} next express next argument
+ * @returns {json} json
+ */
+const editProfile = async (request, response) => {
+  const { body, userData } = request;
+  const {
+    email, firstName, lastName, avatarUrl
+  } = body;
+  if (!email && !firstName && !lastName && !avatarUrl) {
+    return responseMessage(response, 400, { error: 'the field(s) entered are invalid' });
+  }
+  if (email) {
+    const existingUser = await User.findOne({ where: { email } });
+    if (existingUser) return responseMessage(response, 409, { error: 'email already exist' });
+  }
+  const updatedUser = await User.update({ ...body }, {
+    where: { id: userData.id },
+    returning: true,
+    raw: true
+  });
+  const updatedValues = updatedUser[1][0];
+  let following = false;
+  if (request.userData) {
+    const { id } = request.userData;
+    const follower = await findFollower(updatedValues.id, id);
+    if (follower) following = true;
```

Comment:
Is there a way for you to write this better? Do you think `following = follower; ` is a good way to write this considering in this case if there's a `follower`, `following` resolves to `true` and `false` if `follower` is undefined or null?

Reply:
I think the better way might be `following = !!follower` because the response returned is null or an object, using the double negation gives the boolean value of the result